packages/grafana-ui/src/components/Table/TableNG/components/TableCellTooltip.tsx

```diff
@@ -0,0 +1,159 @@
+import { selectors } from '@grafana/e2e-selectors';
+import { TableCellTooltipPlacement } from '@grafana/schema';
+
+import { Popover } from '../../../Tooltip/Popover';
+import { PopoverController } from '../../../Tooltip/PopoverController';
+import { TableCellOptions } from '../../types';
+import { TABLE } from '../constants';
+import { TableCellRenderer, TableCellRendererProps } from '../types';
+
+export interface Props {
+  cellOptions: TableCellOptions;
+  children: ReactElement;
+  className?: string;
+  data: DataFrame;
+  disableSanitizeHtml?: boolean;
+  field: Field;
+  getActions: (field: Field, rowIdx: number) => ActionModel[];
+  height?: number;
+  placement?: TableCellTooltipPlacement;
+  popoverRef: React.MutableRefObject<HTMLElement | null>;
+  renderer: TableCellRenderer;
+  root?: HTMLElement;
+  rowIdx: number;
+  style?: CSSProperties;
+  theme: GrafanaTheme2;
+  tooltipCaretClassName: string;
+  tooltipWrapperClass: string;
+  width?: number;
+}
+
+export function TableCellTooltip({
+  cellOptions,
+  children,
+  className,
+  data,
+  disableSanitizeHtml,
+  field,
+  getActions,
+  height: _height,
+  placement,
+  popoverRef,
+  renderer,
+  root,
+  rowIdx,
+  style,
+  theme,
+  tooltipCaretClassName,
+  tooltipWrapperClass,
+  width = 300,
+}: Props) {
+  const rawValue = field.values[rowIdx];
+  const height = _height ?? TABLE.MAX_CELL_HEIGHT;
+  const tooltipContentRef = useRef<HTMLDivElement>(null);
+
+  const [pinned, setPinned] = useState(false);
+
+  const rendererProps = useMemo(
+    () =>
+      ({
+        cellInspect: false,
+        cellOptions,
+        disableSanitizeHtml,
+        field,
+        frame: data,
+        getActions,
+        height,
+        rowIdx,
+        showFilters: false,
+        theme,
+        value: rawValue,
+        width,
+      }) satisfies TableCellRendererProps,
+    [cellOptions, data, disableSanitizeHtml, field, getActions, height, rawValue, rowIdx, theme, width]
+  );
+
+  if (rawValue === null || rawValue === undefined) {
+    return children;
+  }
+
+  const body = (
+    <div ref={tooltipContentRef} data-testid={selectors.components.Panels.Visualization.TableNG.Tooltip.Wrapper}>
+      {renderer(rendererProps)}
+    </div>
+  );
+
+  return (
+    <PopoverController content={body} placement={placement}>
+      {(_showPopper, _hidePopper, popperProps) => {
+        const showPopper = () => {
+          _showPopper();
+        };
+
+        const hidePopper = () => {
+          if (!pinned) {
+            _hidePopper();
+          }
+        };
+
+        const unpinPopper = () => {
+          setPinned(false);
+          _hidePopper();
+        };
+
+        const pinPopper = (origEvent: SyntheticEvent) => {
+          setPinned(true);
+          _showPopper();
+
+          const origTarget = origEvent.currentTarget;
+
+          const listener = (event: MouseEvent) => {
+            const clickTarget = event.target as Node;
```

Comment:
think you can do something like `if (event.target instanceof Node)` to correctly restrict the type and avoid the type assertion

Reply:
a simpler approach that ended up working was to use `Event` instead of `MouseEvent`. A strange bit of TypeScript trivia, but the types all get along when I do that, so I'm not going to look this gift horse in the mouth.